### ISSUE TITLE
An issue about Object.assign

### ISSUE THREAD

**Issue by NWU-NISL** [mozilla/rhino#783]
###### Version

rhino-1.7.13

###### Test case

```js
var o = {
    get x(){}
}; 
Object. assign(o,o);
```

###### Execution steps

    java -jar rhino-1.7.13.jar -debug -version 200 testcase.js

###### Output

```

```

###### Expected behavior

```js
TypeError: Cannot set property x of [object Object] which has only a getter
```

###### Description

When the source object and target object of `Object.assign` are both objects that only contain getter properties, other engines (such as v8, spiderMonkey, JavaScriptCore, etc.) threw a `TypeError `exception, but rhino passed it normally.

**Comment by tuchida** [comment]
I think this is probably.
https://github.com/mozilla/rhino/blob/41e57c749027ff20b3977a3b3b25e1b7ad6d45df/testsrc/test262.properties#L693
https://github.com/tc39/test262/blob/041da54c02ae7d17edb8c134ab7691c4f643bafe/test/built-ins/Object/assign/target-set-not-writable.js

**Comment by rbri** [comment]
Will have a look

**Comment by tuchida** [comment]
@rbri I fixed it #812. Could you review if you like?

**Comment by gbrail** [comment]
Thanks!
